### PR TITLE
Column and row stabilizers for skew tableaux

### PR DESCRIPTION
**Pull request by darijgr**
Row and column stabilizers and the usual a, b, e elements of the symmetric group algebra are now computable for skew standard tableaux, not just for straight ones.

The a, b, e functions are still not exposed at class level, but this can be done another day.

The performance cost (at least on the e function) is trivial:

```
sage: %timeit eold([[2,3],[1,4]])
1.29 ms ± 12.7 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
sage: %timeit enew([[2,3],[1,4]])
1.29 ms ± 15.5 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
sage: %timeit eold([[2,3],[1,4],[5,7]])
3.43 ms ± 141 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
sage: %timeit enew([[2,3],[1,4],[5,7]])
3.44 ms ± 158 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
(Tested using factored-out code, since the actual functions are cached.)

### CODE REVIEW

File: src/sage/combinat/skew_tableau.py [hunk 1]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
         """
         return [list(row) for row in self]
 
+    def row_stabilizer(self):
+        """
+        Return the PermutationGroup corresponding to the row stabilizer of
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py]:
```suggestion
        Return the :func:`PermutationGroup` corresponding to the row stabilizer of
```

File: src/sage/combinat/skew_tableau.py [hunk 2]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
...
+        gens = [list(range(1, k + 1))]
+        for row in self:
+            for j in range(len(row) - 1):
+                if row[j] is not None:
+                    gens.append((row[j], row[j + 1]))
+        return PermutationGroup(gens)
+
+    def column_stabilizer(self):
+        """
+        Return the PermutationGroup corresponding to the column stabilizer
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py]:
```suggestion
        Return the :func:`PermutationGroup` corresponding to the column stabilizer
```

File: src/sage/combinat/skew_tableau.py [hunk 3]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
         """
         return [list(row) for row in self]
 
+    def row_stabilizer(self):
+        """
+        Return the PermutationGroup corresponding to the row stabilizer of
+        ``self``.
+
+        This assumes that every integer from `1` to the size of ``self``
+        appears exactly once in ``self``.
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py:1149]:
```suggestion
        This assumes that every integer from `1` to the size of ``self``
        appears exactly once in ``self``; i.e., a standard skew tableau.
```

File: src/sage/combinat/skew_tableau.py [hunk 4]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
...
+                    gens.append((row[j], row[j + 1]))
+        return PermutationGroup(gens)
+
+    def column_stabilizer(self):
+        """
+        Return the PermutationGroup corresponding to the column stabilizer
+        of ``self``.
+
+        This assumes that every integer from `1` to the size of ``self``
+        appears exactly once in ``self``.
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py:1192]:
```suggestion
        appears exactly once in ``self``; i.e., a standard skew tableau.
```

File: src/sage/combinat/skew_tableau.py [hunk 5]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
...
+
+        EXAMPLES::
+
+            sage: # needs sage.groups
+            sage: cs = SkewTableau([[None,2,3],[1,5],[4]]).column_stabilizer()
+            sage: cs.order() == factorial(2)*factorial(2)
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py]:
```suggestion
            sage: cs.order() == factorial(2) * factorial(2)
```

File: src/sage/combinat/skew_tableau.py [hunk 6]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
...
+        ``self``.
+
+        This assumes that every integer from `1` to the size of ``self``
+        appears exactly once in ``self``.
+
+        EXAMPLES::
+
+            sage: # needs sage.groups
+            sage: rs = SkewTableau([[None,1,2,3],[4,5]]).row_stabilizer()
+            sage: rs.order() == factorial(3)*factorial(2)
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py]:
```suggestion
            sage: rs = SkewTableau([[None,1,2,3], [4,5]]).row_stabilizer()
            sage: rs.order() == factorial(3) * factorial(2)
```
And perhaps similar spacing between the lists. At least, it should be consistent (see the next `rs = ...` below).

**Review by tscrim — CHANGES_REQUESTED** [review]
Just some minor details.

File: src/sage/combinat/skew_tableau.py [hunk 7]
```diff
@@ -1139,6 +1140,81 @@ def to_list(self):
...
+
+            sage: # needs sage.groups
+            sage: rs = SkewTableau([[None,1,2,3],[4,5]]).row_stabilizer()
+            sage: rs.order() == factorial(3) * factorial(2)
+            True
+            sage: PermutationGroupElement([(1,3,2),(4,5)]) in rs
+            True
+            sage: PermutationGroupElement([(1,4)]) in rs
+            False
+            sage: rs = SkewTableau([[None, 1, 2],[3]]).row_stabilizer()
```
Comment (tscrim) [inline src/sage/combinat/skew_tableau.py]:
```suggestion
            sage: rs = SkewTableau([[None,1,2],[3]]).row_stabilizer()
```
For consistency.

File: src/sage/combinat/symmetric_group_algebra.py [hunk 8]
```diff
@@ -2793,6 +2813,11 @@ def e(tableau, star=0):
         sage: QS3.antipode(e([[1,2],[3]]))
         [1, 2, 3] + [2, 1, 3] - [2, 3, 1] - [3, 2, 1]
 
+    And here is an example for a skew tableau::
+
+        sage: e([[None, 2, 1], [4, 3]])
+        [1, 2, 3, 4] + [1, 2, 4, 3] - [1, 3, 2, 4] - [1, 4, 2, 3] + [2, 1, 3, 4] + [2, 1, 4, 3] - [2, 3, 1, 4] - [2, 4, 1, 3]
```
Comment (tscrim) [inline src/sage/combinat/symmetric_group_algebra.py]:
```suggestion
        [1, 2, 3, 4] + [1, 2, 4, 3] - [1, 3, 2, 4] - [1, 4, 2, 3] 
         + [2, 1, 3, 4] + [2, 1, 4, 3] - [2, 3, 1, 4] - [2, 4, 1, 3]
```
To break the long output.

**Review by tscrim — APPROVED** [review]
Thank you. LGTM.

**Comment by darijgr** [comment]
Thank you, Travis!

**Comment by mkoeppe** [comment]
Lint failure, please fix

**Comment by darijgr** [comment]
Right, good catch. Fixed!

**Pull request merged**